genCode.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from io import open
+
+import json
+import os
+import re
+import six
+import textwrap
+
+from jinja2 import Environment, FileSystemLoader
+
+GENERATED_STRING = "# This file is generated!  Do not edit!"
+# ROUTING_KEY_WHITELIST is also defined in test/base/__init__.py
+# if you change it, please update it there as well.
+ROUTING_KEY_WHITELIST = ("name", "multipleWords", "constant")
+
+
+def loadJson(filename):
+    with open(filename) as fh:
+        return json.load(fh)
+
+
+def createInitPy(pydir, modules):
+    initpy = os.path.join(pydir, '__init__.py')
+    print(initpy)
+    with open(initpy, 'w') as fh:
+        print("#!/usr/bin/env python", file=fh)
+        print(GENERATED_STRING, file=fh)
+        for module in modules:
+            print("from taskcluster.sync.{module} import {module}".format(module=module),
+                  file=fh)
+        print("__all__ = [", file=fh)
+        for module in modules:
+            print("    {module},".format(module=module), file=fh)
+        print("]", file=fh)
+
+
+def createRoutes(api):
+    """Create a string to define self.routes.
+    """
+    routes = ''
+    for entry in api['entries']:
+        if entry['type'] == 'function':
+            routes += "'%s': '%s',\n" % (entry['name'], anglesToBraces(entry['route']))
+    return routes
+
+
+def createRoutingKeys(api):
+    """Create a string to define self.routingKey
+
+    This doesn't include everything in routingKey, because
+        a) only a few things are currently used, and
+        b) for the sake of line and file length
+    """
+    parts = []
+    for entry in api['entries']:
+        if 'routingKey' in entry:
+            parts.append("    '%s': [" % entry['name'])
+            for routingKey in entry['routingKey']:
+                parts.append("        {")
+                for key in ROUTING_KEY_WHITELIST:
+                    if key in routingKey:
+                        value = routingKey[key]
+                        if isinstance(value, (six.binary_type, six.text_type)):
+                            value = "'{}'".format(value)
+                        parts.append("            '%s': %s," % (key, value))
+                parts.append("        },")
+            parts.append("    ],")
+    return '\n'.join(parts)
+
+
+def methodArgumentString(entry):
+    """Returns an argument string for the given function
+    """
+    parts = ['self']
+    parts.extend(argumentNames(entry))
+    if entry.get('query'):
+        parts.append('options=None')
+    return ", ".join(parts)
+
+
+def argumentNames(entry):
+    parts = []
+    if 'args' in entry:
+        parts.extend(entry['args'])
+    if 'input' in entry:
+        inputName = 'payload'
+        parts.append(inputName)
+    return parts
+
+
+def argumentString(entry):
```

Comment:
Why do we need this function at all?

What about:

`def generateArguments(entry, includeSelf=False):`

?